File: backend/backend/Models/SimulationData.cs
using backend.Models;
using Microsoft.EntityFrameworkCore;

namespace backend.Data;
[Owned]
public class SimulationData
{
    private SimulationData() { }

    public SimulationData(
        Party[] parties,
        Dictionary<string, DistrictDetails> districts,
        Dictionary<string, int[]> votesInAreas)
    {
        Parties = parties;
        Districts = districts;
        VotesInAreas = votesInAreas;
    }

    public Party[] Parties { get; set; } = Array.Empty<Party>();

    public Dictionary<string, DistrictDetails> Districts { get; set; }
        = new();

    public Dictionary<string, int[]> VotesInAreas { get; set; }
        = new();
}

File: frontend/Polish-election-simulation/src/api/SimulationData.ts
export interface Party {
  name: string;
  isCoalition: boolean;
  needsThreshold: boolean;
}

export interface  District {
  seats: number;
  terytCodes: string[];
}

export interface SimulationData {
  parties: Party[];
  districts: {
    [key: string]: District;
  };
  votesInAreas: {
    [key: string]: number[];
  };
}